mlir/lib/CAPI/Dialect/Rock.cpp

```diff
@@ -0,0 +1,79 @@
+//===- MIGraphX.cpp - C Interface for MIGraphX dialect---------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir-c/Dialect/Rock.h"
+#include "mlir/CAPI/IR.h"
+#include "mlir/CAPI/Pass.h"
+#include "mlir/CAPI/Registration.h"
+#include "mlir/CAPI/Wrap.h"
+#include "mlir/Dialect/Rock/IR/Rock.h"
+#include "mlir/Dialect/Rock/Tuning/ConvContext.h"
+#include "mlir/Dialect/Rock/Tuning/RockTuning.h"
+
+#include <vector>
+
+MLIR_DEFINE_CAPI_DIALECT_REGISTRATION(Rock, rock, mlir::rock::RockDialect)
+DEFINE_C_API_PTR_METHODS(MlirRockTuningSpace, mlir::rock::TunableParams)
+DEFINE_C_API_PTR_METHODS(MlirRockTuningParam, mlir::rock::ParamEntry)
+// DEFINE_C_API_METHOD(MlirRockTuningKey, const void);
+
+using namespace mlir;
+
+MLIR_CAPI_EXPORTED MlirRockTuningSpace
+mlirRockTuningSpaceCreate(MlirModule module) {
+  struct rock::TunableParams *newParams;
+  auto mod = unwrap(module);
+  newParams = rock::createTunableParams(mod);
+  return wrap(newParams);
+}
+
+MLIR_CAPI_EXPORTED int
+mlirRockTuningGetNumParamsQuick(MlirRockTuningSpace params) {
+  auto tuningSpace = unwrap(params);
+  return tuningSpace->numHeuristicQuick;
+}
```

Comment:
How is this API useful? Or more fundamentally, what does `numHeuristicQuick` field trying to do? I see it is set to 0 now which make me think this field shouldn't exist.

Reply:
This is assuming we can sort the tuning parameters in "likely to optimal" order with an analysis, and also can do another analysis to determine the number of tries until reach at the optimal number.
Currently it's set 0 because we don't have that analysis in current heuristic yet.

Reply:
This sounds like a interesting idea, I think it does worth exploring. I have an question in the following use case:

The client want to do the tuning in the `quick` tuning range as well as using the `numHeuristicQuick` amount of times to do the tuning. However, it looks like `mlirRockTuningSetParam(tuningSpace, 0)` does not seem to care about which tuning space it is in. The results of this seems to indicate that the client is still working at the `full` tuning space, just less in numbers.

I think in order to achieve what you describe, you will need to differentiate between the two sorts of tuning space as well:
 - mlirRockTuningSpaceCreateQuick()
 - mlirRockTunignSpaceCreateFull()

Reply:
create tuning space always returns full range of sorted list. Let's say, full list has 100 entry and numHeuristicQuick=20. Client only needs to try 0~19 params in the tuning space.

Reply:
We can address this later and I can see you can re-arrange the sequence of tuning space to make this work.